perpendicular-flap/solid-fake/fake.py

```diff
@@ -0,0 +1,78 @@
+from __future__ import division
+
+import numpy as np
+import precice
+
+
+def displace_flap(x, y, t, flap_tip_y):
+    x_displ = np.zeros_like(x)
+    y_displ = np.zeros_like(y)
+    # first, get displacement independent of x, only dependent on y and t
+    # maximal displacement at flap tip should be 0.5
+    # initially, flap's displacement is 0
+    x_displ = np.minimum(((np.sin(3 * np.pi * t + np.arcsin(-0.95)) + 0.95) / 8) * y / flap_tip_y, 0.5 * y / flap_tip_y)
```

Comment:
```suggestion
    # first, get displacement independent of x, only dependent on y and t
    # initially, flap's displacement is 0
    max_x_displacement = 0.5
    x_displ = np.minimum(((np.sin(3 * np.pi * t + np.arcsin(-0.95)) + 0.95) / 8) * y / flap_tip_y, max_x_displacement * y / flap_tip_y)
```

I would generally try to avoid raw numbers and rather use descriptive names. This often allows you to completely save some comments.

What does the `0.95`, the `3 * np.pi` and the `/8` do? Generally, I think if the formula you are using here works we can keep it. But a brief comment about the general rationale behind the `arcsin` and `sin` would be helpful.

Reply:
I added some comments that hopefully clarify the idea behind the formula